execution_engine/src/core/engine_state/mod.rs

```diff
@@ -505,6 +492,21 @@ where
             }
         }
 
+        // We insert the new unbonding delay once the purses to be paid out have been transformed
```

Comment:
+1 Good thinking, seems like without this change we'd use the new unbonding delay during migration which could cause funds to get paid earlier, or later than expected.